exod/server/kernel.go

```diff
@@ -77,10 +82,56 @@ func (kern *Kernel) GetVersion(ctx context.Context, input *api.GetVersionInput)
 	}, nil
 }
 
+func (kern *Kernel) Upgrade(ctx context.Context, input *api.UpgradeInput) (*api.UpgradeOutput, error) {
+	upgraded, err := telemetry.TrySelfUpgrade()
+	if err != nil {
+		return nil, err
+	}
+	if upgraded {
+		defer restart(ctx)
+	}
+	return &api.UpgradeOutput{}, nil
+}
+
 func (kern *Kernel) Panic(ctx context.Context, input *api.PanicInput) (*api.PanicOutput, error) {
 	message := input.Message
 	if input.Message == "" {
 		message = "test error"
 	}
 	panic(message)
 }
+
+// restart replaces the current process with a new copy of itself. This is useful
+// after downloading a new binary.
+func restart(ctx context.Context) {
```

Comment:
Generally, it's a tad weird that this is in the "server" package, where most/all of the other CLI functionality is confined to the `cmd` directory of the `exod` package. However, I can't really think of a better way to do this, so I think the layering violation is justified. You might want to do a NOTE or something to cross reference the "server" command, the "upgrade" command, and this code.